crates/vhost/src/vhost_user/message.rs

```diff
@@ -492,8 +506,31 @@ pub struct VhostUserMemoryRegion {
     pub user_addr: u64,
     /// Offset where region starts in the mapped memory.
     pub mmap_offset: u64,
+
+    #[cfg(feature = "xen")]
+    /// Xen specific flags.
+    pub xen_mmap_flags: u32,
+
+    #[cfg(feature = "xen")]
+    /// Xen specific data.
+    pub xen_mmap_data: u32,
+}
+
+impl VhostUserMemoryRegion {
+    fn validate(&self) -> bool {
+        if self.memory_size == 0
+            || self.guest_phys_addr.checked_add(self.memory_size).is_none()
+            || self.user_addr.checked_add(self.memory_size).is_none()
+            || self.mmap_offset.checked_add(self.memory_size).is_none()
+        {
+            return false;
+        }
+
+        true
```

Comment:
just negate the if statement and drop the branching?

```rs
    fn validate(&self) -> bool {
        self.memory_size != 0
            && self.guest_phys_addr.checked_add(self.memory_size).is_some()
            && self.user_addr.checked_add(self.memory_size).is_some()
            && self.mmap_offset.checked_add(self.memory_size).is_some())
    }
```
Maybe one could also call this something like `is_map_info_valid` in order to prevent the confusion of having `is_valid` and `validate`?